knowledge_graph.py

```diff
@@ -0,0 +1,110 @@
+"""Knowledge Graph"""
+from typing import Dict, List, Any
+from transformers import (
+    Text2TextGenerationPipeline,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer)
+from transformers.pipelines.text2text_generation import ReturnType
+from pydantic import Field
+
+class KnowledgeGraph():
+  """Knowledge Graph class"""
+
+  max_new_tokens: int = Field(
+    default=200, description="Maximum number of new tokens to generate."
+  )
+  predictions : List[Dict[str, Any]] = Field(default=[])
+
+  def __init__(
+    self,
+    *args,
+    model_name:str='Babelscape/rebel-large',
+    **kwargs):
+
+    self.model = EntityRelationshipExtractionPipeline(model=model_name, *args, **kwargs)
+
+  def inference(self, sentence: str) -> Dict[str, Any]:
+    """inference"""
+    self.predictions = self.model(sentence)
+
+    return self.predictions
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `KnowledgeGraph` class is missing a docstring for the `__init__` method. It's important to include docstrings for all public methods, including `__init__`, to provide an understanding of their functionality and usage.

```diff
  def __init__(
    self,
    *args,
    model_name:str='Babelscape/rebel-large',
    **kwargs):
+    """
+    Initializes the KnowledgeGraph with a specific model.
+    
+    Args:
+        model_name (str): The name of the model to use for entity relationship extraction.
+        *args: Variable length argument list.
+        **kwargs: Arbitrary keyword arguments.
+    """

    self.model = EntityRelationshipExtractionPipeline(model=model_name, *args, **kwargs)
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->